extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -285,18 +346,24 @@ export class AmpA4A extends AMP.BaseElement {
         })
         // This block returns the ad creative and signature, if available; null
         // otherwise.
+        /**
+         * @return {!Promise<?{creative: !ArrayBuffer, signature: !ArrayBuffer}>}
+         */
         .then(responseParts => {
           checkStillCurrent(promiseId);
+          this.lifecycleReporter_.sendPing('extractCreativeAndSignature');
           return responseParts && this.extractCreativeAndSignature(
               responseParts.bytes, responseParts.headers);
         })
         // This block returns the ad creative if it exists and validates as AMP;
         // null otherwise.
+        /** @return {!Promise<?string>} */
         .then(creativeParts => {
           checkStillCurrent(promiseId);
           if (!creativeParts || !creativeParts.signature) {
             return null;
           }
+          this.lifecycleReporter_.sendPing('validateAdResponse');
```

Comment:
nitpick to have consistent naming: adResponseValidateStart
In general, start/end is good in name to accurately describe the time point, especially for heavy tasks that do take some time to run.